app/serializers/serializers/habitat_cover_serializer.rb

```diff
@@ -10,48 +10,59 @@ def serialize
     @habitat_cover.each do |habitat, translations|
       habitat_cover_item = translations
 
-      # habitat_global_change = Habitat.find_by(name: habitat.to_s).calculate_global_cover_change
       habitat_global_change = Habitat.find_by(name: habitat.to_s).global_change_stat
 
-      if habitat_global_change.nil?
-        change_percentage = '-'
-      else
-        citations = habitat_global_change.global_change_citations.to_a.first
-
-        if habitat_global_change[:percentage_lost].nil?
-          change_percentage = habitat_global_change[:lower_bound_percentage].round().to_s + ' - ' +
-          habitat_global_change[:upper_bound_percentage].round().to_s + '%'
-        else
-          change_percentage = habitat_global_change[:percentage_lost].round().to_s + '%'
-        end
-        baseline_year = habitat_global_change[:baseline_year]
-        recent_year = habitat_global_change[:recent_year]
-      end
+      stats = reformat(habitat_global_change)
 
       habitat_cover_item[:text] =
-        I18n.t(
-          "home.habitat_cover.habitats.#{habitat}.text",
-          change_percentage: change_percentage,
-          baseline_year: baseline_year,
-          recent_year: recent_year,
+      I18n.t(
+        "home.habitat_cover.habitats.#{habitat}.text",
+        change_percentage: stats[:change_percentage],
+        baseline_year: stats[:baseline_year],
+        recent_year: stats[:recent_year],
         )
 
-    if habitat_global_change
       citation = I18n.t("home.habitat_cover.habitats.#{habitat}.citation",
-        citation: citations[:citation],
-        citation_url: citations[:citation_url]
+        citation: stats[:citations].nil? ? stats[:citation] : stats[:citations][:citation]
         )
-    else
-      citation = I18n.t('home.habitat_cover.no_citation')
-    end
 
-      habitat_cover_item[:change_percentage] = change_percentage
+      if stats[:citations]
+        if stats[:citations][:citation_url].nil?
+          citation_url = I18n.t("home.habitat_cover.no_url")
```

Comment:
I'm still getting a broken link when no url is specified, I think you need to do something similar to the Known habitat occurrence component in the country page, changing the yml file

Reply:
Sorry Luca, I forgot to say that I destroyed all instances of `GlobalChangeStat`, then ran the rake task to rebuild my records! Then you should have the modal like this: 
<img width="1128" alt="Screenshot 2020-06-18 at 11 24 43" src="https://user-images.githubusercontent.com/53650048/85009477-63546300-b156-11ea-8b41-b628f0af09ed.png">